pkg/antctl/raw/traceflow/command_test.go

```diff
@@ -153,3 +224,306 @@ func TestParseFlow(t *testing.T) {
 		}
 	}
 }
+
+func TestRunE(t *testing.T) {
+	tcs := []struct {
+		name        string
+		src         string
+		dst         string
+		outputType  string
+		liveTraffic bool
+		droppedOnly bool
+		expected    string
+	}{
+		{
+			name:     "no source",
+			expected: "Please provide source",
+		},
+		{
+			name:     "no destination",
+			src:      srcPod,
+			expected: "Please provide destination",
+		},
+		{
+			name:        "live traffic",
+			liveTraffic: true,
+			expected:    "One of source and destination must be a Pod",
+		},
+		{
+			name:        "dropped-only",
+			src:         srcPod,
+			dst:         IP,
+			droppedOnly: true,
+			expected:    "--dropped-only works only with live-traffic Traceflow",
+		},
+		{
+			name:       "dummy-traceflow-Pod-to-Pod",
+			src:        srcPod,
+			dst:        dstPod,
+			outputType: "yaml",
+			expected: `phase: Succeeded
+source: default/Pod-1
+destination: default/Pod-2
+`,
+		},
+		{
+			name:       "dummy-traceflow-Pod-to-Service",
+			src:        srcPod,
+			dst:        "default/Service",
+			outputType: "yaml",
+			expected: `phase: Succeeded
+source: default/Pod-1
+destination: default/Service
+`,
+		},
+		{
+			name:       "dummy-traceflow-Pod-to-IP",
+			src:        srcPod,
+			dst:        IP,
+			outputType: "json",
+			expected:   "\n  \"phase\": \"Succeeded\",\n  \"source\": \"default/Pod-1\",\n  \"destination\": \"1.1.1.1\"\n",
+		},
+		{
+			name:        "dummy-traceflow-livetraffic",
+			src:         srcPod,
+			liveTraffic: true,
+			outputType:  "yaml",
+			expected: `phase: Succeeded
+source: default/Pod-1
+`,
+		},
+	}
+
+	for _, tt := range tcs {
+		t.Run(tt.name, func(t *testing.T) {
+			modifyCommandAndOption(tt.src, tt.dst, tt.outputType, tt.liveTraffic, tt.droppedOnly, false)
+
+			client := &antreafakeclient.Clientset{}
+			var tf v1alpha1.Traceflow
+			client.AddReactor("create", "traceflows", func(action k8stesting.Action) (bool, runtime.Object, error) {
+				createAction := action.(k8stesting.CreateAction)
+				obj := createAction.GetObject().(*v1alpha1.Traceflow)
+				obj.Status.Phase = v1alpha1.Succeeded
+				tf = *obj.DeepCopy()
+				return true, obj, nil
+			})
+			client.AddReactor("get", "traceflows", func(action k8stesting.Action) (bool, runtime.Object, error) {
+				return true, &tf, nil
+			})
+
+			getClients = func(cmd *cobra.Command) (kubernetes.Interface, antrea.Interface, error) {
+				return k8sClient, client, nil
+			}
+
+			buf := new(bytes.Buffer)
+			Command.SetOutput(buf)
+			Command.SetOut(buf)
+			Command.SetErr(buf)
+			err := runE(Command, nil)
+			require.NoError(t, err)
+			assert.Contains(t, buf.String(), tt.expected)
+			// set Command and option back to their original value.
+			modifyCommandAndOption("", "", "yaml", false, false, false)
+		})
+	}
+	getClients = getK8sClient
```

Comment:
use `defer func() {getClients = getK8sClient}()` after changing `getClient` to ensure the restore code will be executed for all cases